File: generating_face_data/C++/Generating the data for face recognition.cpp
#include <iostream>
#include <opencv.hpp>
#include <filesystem>
#include <sstream>

#if _DEBUG
#pragma comment(lib, "opencv_world480d.lib")
#else
#pragma comment(lib, "opencv_world480.lib")
#endif

using namespace cv;
using namespace std;

int main()
{
    string output_folder_base = "../data/";
    string folder_name = "Mosiow";
    std::filesystem::path output_folder = std::filesystem::path(output_folder_base) / folder_name;
    if (!std::filesystem::exists(output_folder)) {
        std::filesystem::create_directories(output_folder);
    }

    CascadeClassifier face_cascade;
    bool load = face_cascade.load("D:/Internship/Files needed/haarcascade/haarcascade_frontalface_default.xml");
    VideoCapture camera(0);
    int count = 0;
    while (count <= 100)
    {
        Mat frame;
        bool success = camera.read(frame);
        if (success)
        {
            Mat gray;
            cvtColor(frame, gray, COLOR_BGR2GRAY);
            vector<Rect> faces;
            Size minSize(120, 120);
            face_cascade.detectMultiScale(gray, faces, 1.3, 5, 0, minSize);
            for (const Rect& face : faces)
            {
                int x = face.x;
                int y = face.y;
                int w = face.width;
                int h = face.height;
                rectangle(frame, Point(x, y), Point(x + w, y + h), (255, 0, 0), 2);
                Mat face_img;
                Size s(200, 200);
                resize(gray(Rect(x, y, w, h)), face_img, s);
                imshow("Capturing Faces...", frame);
                waitKey(1);

                std::filesystem::path face_filename = output_folder / (folder_name + "_" + std::to_string(count) + ".pgm");
                string face_filename_str = face_filename.string();
                imwrite(face_filename_str, face_img);
                count += 1;
            }
        }
    }
    destroyAllWindows();
}

File: generating_face_data/python/Generating_the_data_for_face_recognition.py
import cv2
import os

output_folder_base = '../data/'
folder_name = 'AbbasAli'
output_folder = os.path.join(output_folder_base, folder_name)
if not os.path.exists(output_folder):
    os.makedirs(output_folder)

face_cascade = cv2.CascadeClassifier('D:\Internship\Files needed\haarcascade\haarcascade_frontalface_default.xml')
camera = cv2.VideoCapture(0)

count = 0
while (count <= 100):

    success, frame = camera.read()
    if success:
        gray = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
        faces = face_cascade.detectMultiScale(gray, 1.3, 5, minSize=(120, 120))
        for (x, y, w, h) in faces:
            cv2.rectangle(frame, (x, y), (x+w, y+h), (255, 0, 0), 2)
            face_img = cv2.resize(gray[y:y+h, x:x+w], (200, 200))
            cv2.imshow('Capturing Faces...', frame)
            cv2.waitKey(1) 
            face_filename = '%s/%s_%d.pgm' % (output_folder, folder_name, count)
            cv2.imwrite(face_filename, face_img)
            count += 1

cv2.destroyAllWindows()